src/algebra/algebra/basic.lean

```diff
@@ -1291,13 +1291,38 @@ def to_int_alg_hom [ring R] [ring S] [algebra ℤ R] [algebra ℤ S] (f : R →+
   (f : R →+* S) (r : ℚ) : f (algebra_map ℚ R r) = algebra_map ℚ S r :=
 ring_hom.ext_iff.1 (subsingleton.elim (f.comp (algebra_map ℚ R)) (algebra_map ℚ S)) r
 
-/-- Reinterpret a `ring_hom` as a `ℚ`-algebra homomorphism. -/
+/-- Reinterpret a `ring_hom` as a `ℚ`-algebra homomorphism. This actually yields an equivalence,
+see `ring_hom.equiv_rat_alg_hom`. -/
 def to_rat_alg_hom [ring R] [ring S] [algebra ℚ R] [algebra ℚ S] (f : R →+* S) :
   R →ₐ[ℚ] S :=
 { commutes' := f.map_rat_algebra_map, .. f }
 
+@[simp]
+lemma rat_alg_hom_to_ring_hom_eq_id [ring R] [ring S] [algebra ℚ R] [algebra ℚ S]
+(f : R →+* S) : f.to_rat_alg_hom.to_ring_hom = f :=
+ring_hom.ext (by {intro x, refl, })
+
 end ring_hom
 
+section
+
+variables {R S : Type*}
+
+@[simp]
+lemma alg_hom.ring_hom_to_rat_alg_hom_eq_id [ring R] [ring S] [algebra ℚ R] [algebra ℚ S]
+(f : R →ₐ[ℚ] S) : f.to_ring_hom.to_rat_alg_hom = f :=
+alg_hom.ext (by {intro x, refl, })
+
+@[simps]
+def ring_hom.equiv_rat_alg_hom [ring R] [ring S] [algebra ℚ R] [algebra ℚ S] :
+(R →ₐ[ℚ] S) ≃ (R →+* S) :=
```

Comment:
The name suggests that this should be the other way around
```suggestion
  (R →+* S) ≃ (R →ₐ[ℚ] S) :=
```
(You'll need to swap the fields below too)